internal/olm/operator/install_mode.go

```diff
@@ -93,15 +99,44 @@ func (i InstallMode) Validate() error {
 
 // CheckCompatibility checks if an InstallMode is compatible with the operator's namespace and is supported by csv.
 func (i InstallMode) CheckCompatibility(csv *v1alpha1.ClusterServiceVersion, operatorNamespace string) error {
+	// allnamespace was validated in Validate()
+
+	// own namespace and targetns != opname
 	if i.InstallModeType == v1alpha1.InstallModeTypeOwnNamespace {
-		if i.TargetNamespaces[0] != operatorNamespace {
+		if len(i.TargetNamespaces) > 0 && i.TargetNamespaces[0] != operatorNamespace {
 			return fmt.Errorf("install mode %s must match operator namespace %q", i, operatorNamespace)
 		}
 	}
+
+	// single namespace and targetns == opname
+	if i.InstallModeType == v1alpha1.InstallModeTypeSingleNamespace {
+		if len(i.TargetNamespaces) < 1 || i.TargetNamespaces[0] == operatorNamespace {
+			return fmt.Errorf("use install mode %q to watch operator's namespace %q", v1alpha1.InstallModeTypeOwnNamespace, i.TargetNamespaces[0])
+		}
+	}
```

Comment:
I think using the controller's namespace with `SingleNamespace` is valid. However if we are going to perform this check, move it (and the one above it) below the check for support (line 123) so we make sure the mode is supported first.

Reply:
I don't think it is. Developing `kubectl operator` I was noticing some behavior around this that I thought was odd, but I confirmed with @njhale that OLM requires an operator to support `OwnNamespace` if it is watching the namespace in which it is deployed.

Reply:
Also, it looks like this is actually checking two different cases, right?
* `len(i.TargetNamespaces) < 1` should be checked in `Validate()` and error with "SingleNamespace requires exactly one target namespace"
* `i.TargetNamespaces[0] == operatorNamespace` is the piece that goes with the error message you return here.

@jmrodri I know we talked about this a bit last week. Can you remind me if (and why) we settled on keeping the check for OwnNamespace having a target namespace list equal to the install namespace, which therefore requires us to set that correctly before calling this check?

I'm kind of thinking that it makes more sense to check `OwnNamespace` has an empty target namespace list (which we should do in `Validate()`, and only when we are creating the OperatorGroup in the install namespace do we make the inference that `OwnNamespace` means to use the install namespace. It also seems logical at that point because we _must_ align the `OperatorGroup.metadata.namespace` with the `OperatorGroup.spec.targetNamespaces`. WDYT?